app/src/main/shared/services/AbstractLocalService.ts

```diff
@@ -87,28 +97,54 @@ export abstract class LocalService {
 
     public get(url: string){
         console.log('get', url);
-        return this.doFetch(this.makeUrl(url), this.makeRequestOptions('GET'))
-            .then(this.processResponse)
-            .catch(this.notifyOnErrors);
+        return this.getData(url, "GET");
     }
 
     public post(url: string, params?:any){
         console.log('post', url, params);
-        return this.doFetch(this.makeUrl(url), this.makeRequestOptions('POST', params))
+        return this.getData(url, "POST", params);
+    }
+
+    protected getCache(url: string) {
+        return this.cacheEngine.get(["localService", this.constructor.name, this.options.cache, encodeURIComponent(url)].join('.'));
+    }
+
+    protected setCache(url: string, data: any) {
+        this.cacheEngine.set(["localService", this.constructor.name, this.options.cache, encodeURIComponent(url)].join('.'), data);
+    }
+
+    public clearAllCache() {
+        this.cacheEngine.clearAll();
+        return this;
+    }
+
+    protected getData(url: string, method: string, params?: any) {
+        if (this.options.cache) {
+            const cacheValue = this.getCache(this.makeUrl(url));
+            if (cacheValue) {
+                // reset options on returning cached data from endpoint
+                this.initOptions();
+                return Promise.resolve(cacheValue);
+            }
+        }
+        return this.doFetch(this.makeUrl(url), this.makeRequestOptions(method, params))
```

Comment:
`this.makeUrl(url)` is possibly called twice in this method now, (line 123 and here) so could pull it out into a local variable